tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -84,11 +85,7 @@ sealed trait NativeConfig {
    *  architecture for a name that is not found seems excessive perhaps?
    */
   def is32BitPlatform = {
-    targetTriple
-      .getOrElse(detectedTriple)
-      .split('-')
-      .headOption
-      .getOrElse("") match {
+    configuredOrDetectedTriple.arch match {
```

Comment:
Is there a way to determine 32 bit via the ported code or maybe we could match the known 32 bit and like 386 etc. and `contains("32")` and let the default be 64 bit?

Reply:
> let the default be 64 bit?

That is currently the default. The difference is whether we print a warning when we default to 64-bit because we are defaulting, not because we actually know it's 64 bit.

Reply:
Well, now that you have the entire list you could Unknown warn to 64bit if you can't find an arch and then list the known 32bit ones and also find ones that have 32 in the name so it has a chance of working on all known processors.

Reply:
Overall, what I am trying to say is that this feature is really for the 32bit support with the normal case being 64bit. So we could use this as the basic approach.

1. If it is a 32 bit arch and found in the list then return 32 bit (my idea was x86 plus ones with 32 in the name unless that is a bad assumption)
2. If it is a 64 bit arch and found then return 64 bit
3. If it is not found, assume 64 bit.

Note 1: If it is a 32bit os running on 64 bit then the user needs to supply a target triple.
Note 2: We most likely don't expect to add that many 32bit so hopefully maintaining a list should not be hard where needed.

The Swift code has some code for pointer width / 32bit - https://github.com/apple/swift-driver/blob/main/Sources/SwiftDriver/Utilities/Triple.swift